community/kernel/src/main/java/org/neo4j/helpers/SortedLongArrayUtil.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.helpers;
+
+/**
+ * Specialized methods for operations of sorted long arrays.
+ */
+public class SortedLongArrayUtil
+{
+    private static final long[] EMPTY = new long[]{};
+
+    /**
+     * Compute union of two sorted long arrays.
+     * @param left a sorted array set
+     * @param right another sorted array set
+     * @return the union, which is NOT sorted
+     */
+    public static long[] union( long[] left, long[] right )
```

Comment:
This looks very similar to what I recently introduced in [`PrimitiveSortedArraySet`](https://github.com/neo4j/neo4j/blob/3.2/community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveSortedArraySet.java#L35), how about you add this method there instead? Possibly implementing the same algorithm as the method that is already there (but with `long[]` in the signature instead of `int[]`) so that the method is composable.
This implementation not returning a sorted set means that it has to be sorted externally in order to use it as input in a later iteration, and I don't think that returning sorted output adds to the complexity of the implementation. In fact if you compare the two implementations, my sorted version has _lower_ time complexity (although slightly higher memory consumption in the worst case).

At the very least, I think this class belongs better in the `neo4j-primitives` component, since afaik the `helpers` package is being phased out.